application/app/javascript/mixins/spinner_toggle.js

```diff
@@ -0,0 +1,40 @@
+export default function SpinnerToggleMixin(Base) {
+    return class extends Base {
+        static targets = ["spinner", "label"]
+
+        showSpinner(buttonElement = null) {
+            const button = buttonElement || this.element.querySelector("button")
+            if (!button) return
+
+            const width = button.offsetWidth
+            const height = button.offsetHeight
+            button.style.width = `${width}px`
+            button.style.height = `${height}px`
+            button.disabled = true
```

Comment:
adding or removing CSS classes in the components is advised rather than injecting inline style properties. Can this feature be implemented with CSS classes instead?

Reply:
No - same as before. The total width and height of the button is dynamic and depends on the text and icons